opal/mca/threads/qthreads/configure.m4

```diff
@@ -60,6 +114,21 @@ AC_DEFUN([MCA_opal_threads_qthreads_POST_CONFIG],[
            AC_DEFINE_UNQUOTED([MCA_threads_wait_sync_base_include_HEADER],
                               ["opal/mca/threads/qthreads/threads_qthreads_wait_sync.h"],
                               [Header to include for wait_sync implementation])
+           THREAD_CFLAGS="$TPKG_CFLAGS"
+           THREAD_FCFLAGS="$TPKG_FCFLAGS"
+           THREAD_CXXFLAGS="$TPKG_CXXFLAGS"
+           THREAD_CPPFLAGS="$TPKG_CPPFLAGS"
+           THREAD_CXXCPPFLAGS="$TPKG_CXXCPPFLAGS"
+           THREAD_LDFLAGS="$TPKG_LDFLAGS"
+           THREAD_LIBS="$TPKG_LIBS"
+           AC_SUBST(THREAD_CFLAGS)
+           AC_SUBST(THREAD_FCFLAGS)
+           AC_SUBST(THREAD_CXXFLAGS)
+           AC_SUBST(THREAD_CPPFLAGS)
+           AC_SUBST(THREAD_LDFLAGS)
+           AC_SUBST(THREAD_LIBS)
```

Comment:
Same as above.

I didn't check the `threads` framework, but it's a little odd that a component is responsible for AC_SUBST'ing `TPKG_*` and `THREAD_*` values.  Specifically:

1. Are both `TPKG_*` and `THREAD_*` used by the framework?
1. If so, why isn't the framework doing the `AC_SUBST` (vs. the components)?

Reply:
I moved `AC_SUBST()` outside `AS_IF()`.

This implementation is basically a copy from #7675.

1. `TPKG_*` is used only in `opal/mca/thread`. `THREAD_` is used in the top `configure.ac` (https://github.com/open-mpi/ompi/blob/master/configure.ac#L1114-L1117).
2. I believe it is to propagate this "thread" information to the top `configure`, but I do not fully understand the modularity/design of Open MPI. @hppritcha would you have any idea about this?

Reply:
> This implementation is basically a copy from #7675.

Understood.  I guess this was missed in that PR.  ☹️ 

I guess my question still stands: if `TPKG_*` and `THREAD_*` are being used by the framework (and higher -- all the way up in the top-level `configure.ac`), why is it up to the component to `AC_SUBST` them?  That seems weird.

Can you add a commit on there to remove the `AC_SUBSTs` of these values from the components, and instead, do them in `opal/mca/threads/configure.m4`?

Reply:
I added a commit, which fixed the issue (I also rebased my branch to the master):
- `TPKG_*` are not `AC_SUBSTs`; they appear only in `opal/mca/threads/<THREAD_PKG_NAME>/configure.m4`.
- `THREAD_*` are `AC_SUBSTs`. in `opal/mca/threads/configure.m4`.

As far as I checked, there is no compilation issue with Pthreads, Qthreads, and Argobots.